subprojects/language-jvm/src/main/java/org/gradle/api/internal/tasks/compile/daemon/AbstractDaemonCompiler.java

```diff
@@ -72,6 +77,16 @@ public WorkResult execute(T spec) {
         }
     }
 
+    protected BaseForkOptions mergeForkOptions(BaseForkOptions left, BaseForkOptions right) {
+        BaseForkOptions merged = new BaseForkOptions();
+        merged.setMemoryInitialSize(mergeHeapSize(left.getMemoryInitialSize(), right.getMemoryInitialSize()));
+        merged.setMemoryMaximumSize(mergeHeapSize(left.getMemoryMaximumSize(), right.getMemoryMaximumSize()));
+        Set<String> mergedJvmArgs = normalized(left.getJvmArgs());
+        mergedJvmArgs.addAll(normalized(right.getJvmArgs()));
+        merged.setJvmArgs(Lists.newArrayList(mergedJvmArgs));
```

Comment:
This is the way it worked before I made any changes - I simply changed it back to work as it always had.  Whether that is correct or not is debatable.  I suspect there may be some history here that led to the merged options (possibly a backwards compatibility consideration).  I think we probably need to understand better why we did this in the first place before we consider making a breaking change to remove it.